static/app/components/draggableTabs/draggableTabList.tsx

```diff
@@ -24,6 +38,202 @@ interface BaseDraggableTabListProps extends DraggableTabListProps {
+    const availableWidth = outerWidth - addViewTempTabWidth;
+
+    let totalWidth = 0;
+    const overflowing: Node<DraggableTabListItemProps>[] = [];
+
+    for (let i = 0; i < tabsDimensions.length; i++) {
+      totalWidth += tabsDimensions[i].width + 1; // 1 extra pixel for the divider
+      if (totalWidth > availableWidth + 1) {
+        overflowing.push(persistentTabs[i]);
+      }
+    }
+
+    return overflowing.filter(defined);
+  }, [outerWidth, addViewTempTabWidth, persistentTabs, tabsDimensions]);
+
+  return {
+    overflowingTabs,
+    setTabElements,
+    outerRef,
+    addViewTempTabRef,
+    persistentTabs,
+  };
+}
+
+function OverflowMenu({
+  state,
+  overflowTabs,
+}: {
+  overflowTabs: Node<DraggableTabListItemProps>[];
+  state: TabListState<any>;
+}) {
+  const options = useMemo(() => {
+    return overflowTabs.map(tab => {
+      return {
+        value: tab.key,
+        label: tab.textValue,
+        textValue: tab.textValue,
+      };
+    });
+  }, [overflowTabs]);
+
+  return (
+    <CompactSelect
+      options={options}
+      multiple={false}
+      value={state.selectionManager.firstSelectedKey?.toString()}
+      onChange={opt => state.setSelectedKey(opt.value)}
+      position="bottom-end"
+      size="sm"
+      offset={4}
+      trigger={triggerProps => (
+        <OverflowMenuTrigger
+          {...triggerProps}
+          size="sm"
+          borderless
+          showChevron={false}
+          icon={<IconEllipsis />}
+          aria-label={t('More tabs')}
+        />
+      )}
+    />
+  );
+}
+
+function Tabs({
+  orientation,
+  ariaProps,
+  state,
+  className,
+  onReorder,
+  tabVariant,
+  setTabRefs,
+  tabs,
+  overflowingTabs,
+}: {
+  ariaProps: AriaTabListOptions<DraggableTabListItemProps>;
+  onReorder: (newOrder: Node<DraggableTabListItemProps>[]) => void;
+  orientation: 'horizontal' | 'vertical';
+  overflowingTabs: Node<DraggableTabListItemProps>[];
+  setTabRefs: Dispatch<SetStateAction<Array<HTMLDivElement | null>>>;
+  state: TabListState<DraggableTabListItemProps>;
+  tabs: Node<DraggableTabListItemProps>[];
+  className?: string;
+  disabled?: boolean;
+  onChange?: (key: string | number) => void;
+  tabVariant?: BaseTabProps['variant'];
+  value?: string | number;
+}) {
+  const tabListRef = useRef<HTMLUListElement>(null);
+  const {tabListProps} = useTabList({orientation, ...ariaProps}, state, tabListRef);
+
+  const values = useMemo(() => [...state.collection], [state.collection]);
+
+  const isDragging = useIsDragging(tabListRef);
+
+  // Only apply this while dragging, because it causes tabs to stay within the container
+  // which we do not want (we hide tabs once they overflow
+  const dragConstraints = isDragging ? tabListRef : undefined;
+
+  return (
+    <TabListWrap {...tabListProps} className={className} ref={tabListRef}>
+      <ReorderGroup axis="x" values={values} onReorder={onReorder} as="div">
+        {tabs.map((item, i) => (
```

Comment:
I noticed that tab's that are calculated as overflowing have opacity=0 applied to them, which might be the reason why the `+ Add View` button isn't snapping to the new edge when the tab disappears

![image](https://github.com/user-attachments/assets/08aa6371-55ba-4d56-85d1-5698e7ddbcaf)

Would removing tabs from the DOM if they're overflowing solve this?

Reply:
The problem is that we need to render the tabs to know how big they are. Otherwise we won't know when they are able to fit in the available space. I'm sure there is a way to get this behavior, but it would certainly complicate the solution